### ISSUE TITLE
Convert self_cell struct back into owner

### ISSUE THREAD

**Issue by Flakebi** [Voultapher/self_cell#13]
Hi,
rental has a way to convert a struct back into the owner (I think it’s calling `into_head()`). This can be used to e.g. reuse a `Vec<u8>` after the self-referencing struct is not needed anymore.
Could self_cell add something similar?
I hope implementing something like `impl From<$StructName> for $Owner {}` is safe to do.

**Comment by Voultapher** [comment]
Hi, I've thought about something like `into_owner(self) -> $Owner` before. And it should be possible. A tricky question how to do this in a way that doesn't introduce a branch into the drop impl. But maybe a mem::forget solution in into_owner could work.

Regarding your second question:
> I hope implementing something like impl From<$StructName> for $Owner {} is safe to do.

The library is designend such that literally every kind of use and interaction not marked `unsafe` should be safe and sound. If you discover a flaw please report it.

**Comment by Voultapher** [comment]
@Flakebi I've pushed a [commit](https://github.com/Voultapher/self_cell/commit/33d2a7d8ae0bc950a0193bb60506f9e08ba1163f) please let me know if this addresses your use case. You can try out the new version by referencing it like this in your `Cargo.toml`: 

```toml
[dependencies]
self_cell = { git = "https://github.com/Voultapher/self_cell" }
```

**Comment by Flakebi** [comment]
Thanks, that is exactly what I need!

**Comment by Voultapher** [comment]
The newly released version 0.9.0 https://github.com/Voultapher/self_cell/releases/tag/v0.9.0 contains `into_owner`
